src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -125,20 +167,45 @@ public void claimTask(String taskId,
      * @param taskId                the task id.
      * @param accessControlResponse the access control response containing user id and role assignments.
      */
+    @Transactional
     public void unclaimTask(String taskId, AccessControlResponse accessControlResponse) {
         String userId = accessControlResponse.getUserInfo().getUid();
         List<PermissionTypes> permissionsRequired = singletonList(MANAGE);
-        CamundaTask camundaTask = camundaService.getUnmappedCamundaTask(taskId);
-        Map<String, CamundaVariable> variables = camundaService.getTaskVariables(taskId);
+        boolean taskHasUnassigned;
+        final boolean isFeatureEnabled = launchDarklyFeatureFlagProvider
+            .getBooleanValue(FeatureFlag.RELEASE_2_ENDPOINTS_FEATURE,
+                             accessControlResponse.getUserInfo().getUid());
+        if (isFeatureEnabled) {
+            TaskResource taskResource = roleAssignmentVerificationWithAssigneeCheckAndHierarchy(
+                taskId,
+                userId,
+                accessControlResponse,
+                permissionsRequired);
+            String taskState = taskResource.getState().getValue();
+            taskHasUnassigned = taskState.equals(CFTTaskState.UNASSIGNED.getValue());
 
-        roleAssignmentVerificationWithAssigneeCheckAndHierarchy(
-            camundaTask.getAssignee(),
-            userId,
-            variables,
-            accessControlResponse.getRoleAssignments(),
-            permissionsRequired
-        );
-        camundaService.unclaimTask(taskId, variables);
+            //Lock & update Task
+            TaskResource task = findByIdAndObtainLock(taskId);
+            task.setState(CFTTaskState.UNASSIGNED);
+            //Perform Camunda updates
```

Comment:
The assignee should be set to null? If it is unassigned.

In all possible cases, the assignee should be set/reset.